src/pipelines.js

```diff
@@ -272,11 +362,11 @@ class TextGenerationPipeline extends Pipeline {
             inputs_attention_mask: attention_mask
         });
 
-        let toReturn = outputTokenIds.map((outTokens, i) => {
+        let toReturn = outputTokenIds.map((/** @type {any} */ outTokens, /** @type {string | number} */ i) => {
             let startText = texts[i].trim();
             let decoded = this.tokenizer.batch_decode(outTokens, {
                 skip_special_tokens: true,
-            }).map(x => {
+            }).map((/** @type {any} */ x) => {
```

Comment:
Same here, can you declare the return type for `batch_decode`, so `map` can infer the type without any hints?